src/org/zaproxy/zap/extension/domxss/TestDomXSS.java

```diff
@@ -510,15 +519,27 @@ public void scan() {
 					}
 				}
 			}
+			super.scan();
 		} finally {
 			this.returnFirefoxDriver(fxDriver);
 		}
 	}
 
-	public boolean scan(ArrayList<WebDriverWrapper> drivers, String attackVector) {
-    	HttpMessage msg = getBaseMsg();
-    	String url = msg.getRequestHeader().getURI().toString();
-    	String currURL = url + attackVector;
+	public boolean scan(ArrayList<WebDriverWrapper> drivers, String paramName, String attackVector, String uri) {
+		String currURL;
+		HttpMessage msg = getNewMsg();
+		if (!uri.isEmpty()) {
```

Comment:
I was thinking the caller would do this themselves (so here we just used the `currURL`), it would be more flexible for future use cases, fine as is though.

Reply:
I’ll look into that, it does make sense and will require less passing of variables.

Reply:
OK.